wtransport/src/endpoint.rs

```diff
@@ -169,17 +169,46 @@ impl Endpoint<endpoint_side::Server> {
         })
     }
 
+    /// Get the number of connections that are currently open.
+    pub fn open_connections(&self) -> usize {
+        self.endpoint.open_connections()
+    }
+
     /// Get the next incoming connection attempt from a client.
     pub async fn accept(&self) -> IncomingSession {
-        let quic_connecting = self
-            .endpoint
-            .accept()
-            .await
-            .expect("Endpoint cannot be closed");
-
-        debug!("New incoming QUIC connection");
+        self.accept_inner(Some).await
+    }
 
-        IncomingSession::new(quic_connecting)
+    /// Get the next incoming connection attempt from a client.
+    ///
+    /// Applies a `filter` function that may `refuse` or initiate a `retry`,
+    /// such as to mitigate denial of service. Incoming connections that are
+    /// returned will be accepted.
+    #[cfg(feature = "quinn")]
+    pub async fn accept_filter(
+        &self,
+        filter: impl FnMut(quinn::Incoming) -> Option<quinn::Incoming>,
+    ) -> IncomingSession {
+        self.accept_inner(filter).await
+    }
```

Comment:
Previously, one could use `quinn`'s `max_concurrent_connections` and `use_retry` to mitigate DoS, but those were removed.

I experimented with re-implementing them in `wtransport` here: https://github.com/finnbear/wtransport/commit/5dad5aac94a151047602e4a4e36603ff2ea2390d

But `accept_filter` is much more flexible:
- can decide based on IP address
- can randomly `retry` based on load i.e. https://www.ietf.org/archive/id/draft-ietf-quic-retry-offload-00.html